### ISSUE TITLE
gmic_libc.cpp: is_interleaved = true results in mixed attributes

### ISSUE THREAD

**Issue by boessu** [dtschump/gmic-community#112]
I'm afraid I found another bug in the C interface. Hopefully this is the last one. ;-)

If I load an image with the c library non-interleaved, the attributes are just as expected. 
Example  with a jpg image as script input (e.g. "image.jpg"), is_interleaved = false:
> Image format: 0
> Image width: 640
> Image height: 428
> Image depth: 1
> Image spectrum: 3

Example  with the same jpg image is_interleaved = true:
> Image format: 0
**> Image width: 3**
**> Image height: 640**
**> Image depth: 428**
**> Image spectrum: 1**

As you see the value "width" is in "height", "height" is in "depth", "depth" is in "spectrum" and the value of "spectrum" is in "width".

So what I see in the sourcecode is the following if the image is interleaved on the input:
```
        if (_images[i].is_interleaved) {
          img.assign((float*)_images[i].data, _images[i].spectrum, _images[i].width,
                     _images[i].height, _images[i].depth, true);
          img.permute_axes("YZCX");
        } else {
          img.assign((float*)_images[i].data, _images[i].width, _images[i].height,
                     _images[i].depth, _images[i].spectrum, true);
        }
```

But on the output, we don't see any swapped parameters:
```
      gmic_image<float>& img = images[i];
      if (_options && _options->interleave_output) {
        img.permute_axes("CXYZ");
				_images[i].interleave = true;
      } else {
				_images[i].interleave = false;
			}
      if (_options && _options->output_format == E_FORMAT_BYTE) {
        gmic_image<unsigned char> img_tmp;
        img_tmp = img;
        _images[i].format = E_FORMAT_BYTE; 
        _images[i].data = img_tmp._data;
        img_tmp._is_shared = true;
        img._is_shared = false;
      } else {
        _images[i].format = E_FORMAT_FLOAT; 
        _images[i].data = img._data;
        img._is_shared = true;
      }
      _images[i].width = img._width;
      _images[i].height = img._height;
      _images[i].depth = img._depth;
      _images[i].spectrum = img._spectrum;
      std::strcpy(_images[i].name, images_names[i]);
```
So my guess is that the swap of attributes in the interleaving case which will be done on the input image should also be done on the resulting output image.

**Comment by tobybear** [comment]
Great, thanks for your feedback!
I added another fix to the file and uploaded it to the repository.

**Comment by boessu** [comment]
Fixed in release 2.1.9.
